ui/pages/confirmations/components/simulation-details/formatAmount.ts

```diff
@@ -0,0 +1,85 @@
+import { BigNumber } from 'bignumber.js';
+import {
+  MIN_AMOUNT,
+  MIN_AMOUNT_DECIMALS,
+} from '../../../../hooks/useCurrencyDisplay';
+
+// The number of significant decimals places to show for amounts less than 1.
+const MAX_SIGNIFICANT_DECIMAL_PLACES = 3;
+
+const ZERO_DISPLAY = '0';
+
+export const formatAmountMaxPrecision = (
+  locale: string,
+  num: number | BigNumber,
+): string =>
+  new Intl.NumberFormat(locale, {
+    minimumSignificantDigits: 1,
+  }).format(new BigNumber(num.toString()).toNumber());
```

Comment:
Is it safe for us to convert to a number here?

If it is, do we need to be using `BigNumber.js` at all?
If it isn't, what happens if the number is too large for a standard number?